docs/reference/eql/detect-threats-with-eql.asciidoc

```diff
@@ -0,0 +1,427 @@
+later load the `scrob.dll` file.
+
+Modify the previous EQL query as follows:
+
+* Change the `any` keyword to `file`. This means the query only matches
+events with an `event.category` of `file`.
+* Add the `file.name == "scrobj.dll` condition to check for loads of
+the `scrobj.dll` file.
+
+[source,console]
+----
+GET /sec_logs/_eql/search
+{
+  "query": """
+    file where process.name == "regsvr32.exe" and file.name == "scrobj.dll"
+  """,
+  "size": 150
+}
+----
+// TEST[setup:sec_logs]
+
+The query matches an event, confirming that `scrobj.dll` was later loaded by
+`regsvr32.exe`.
+
+[source,console-result]
+----
+{
+  "is_partial": false,
+  "is_running": false,
+  "took": 5,
+  "timed_out": false,
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "events": [
+      {
+        "_index": "sec_logs",
+        "_id": "ol5MJXMBMk1dGnErnBW8",
+        "_score": null,
+        "_source": {
+          "process": {
+            "name": "regsvr32.exe",
+            "pid": 2012,
+            "entity_id": "{42FC7E13-CBCB-5C05-0000-0010A0395401}",
+            "executable": "C:\\Windows\\System32\\regsvr32.exe"
+          },
+          "file": {
+            "path": "C:\\Windows\\System32\\scrobj.dll",
+            "name": "scrobj.dll"
+          },
+          "@timestamp": 131883573237450016,
+          "event": {
+            "category": "file"
+          }
+        },
+        "sort": [
+          131883573237450016
+        ]
+      }
+    ]
+  }
+}
+----
+// TESTRESPONSE[s/"took": 5/"took": $body.took/]
+// TESTRESPONSE[s/"_id": "ol5MJXMBMk1dGnErnBW8"/"_id": $body.hits.events.0._id/]
+
+[discrete]
+[[eql-ex-detemine-likelihood-of-sucess]]
+=== Determine likelihood of success
+
+In many cases, malicious scripts are used to download other remote files. If
+this occurred, the attack progressed and might have succeeded.
+
+Use an <<eql-sequences,EQL sequence query>> to check for the following series of
+events, in order:
+
+. A `regsvr32.exe` process, which could have been used to register malicious
+scripts as `scrobj.dll`
+. A load of the `scrobj.dll` file
+. Any network event, which could indicate the download of a remote file
+
+To match, each event in the sequence must share the same `process.pid` (process
+ID) field value.
+
+[source,console]
+----
+GET /sec_logs/_eql/search
+{
+  "query": """
+    sequence by process.pid
+      [process where process.name == 'regsvr32.exe']
+      [file where file.name == 'scrobj.dll']
+      [network where true]
+  """,
+  "size": 150
+}
+----
+// TEST[setup:sec_logs]
+
+The query matches a sequence, indicating the attack likely succeeded.
+
+[source,console-result]
+----
+{
+  "is_partial": false,
+  "is_running": false,
+  "took": 25,
+  "timed_out": false,
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "sequences": [
+      {
+        "join_keys": [
+          "2012"
```

Comment:
@costin does this always get turned into a string?

Reply:
I believe so - it looks like a bug. Raised https://github.com/elastic/elasticsearch/issues/59707